index.js

```diff
@@ -292,6 +303,46 @@ class Ky {
 
 		return timeout(fetch(this._input, this._options), this._timeout, this.abortController);
 	}
+
+	/* istanbul ignore next */
+	_stream(response, onProgress) {
+		const total = Number(response.headers.get('content-length')) || 0;
+		let transferred = 0;
+
+		return new Response(
+			new ReadableStream({
+				start(controller) {
+					const reader = response.body.getReader();
+
+					if (onProgress) {
+						onProgress(0, 0, total);
+					}
+
+					read();
+					async function read() {
+						const {done, value} = await reader.read();
+						if (done) {
+							onProgress(1, transferred, total);
+							controller.close();
+							return;
+						}
+
+						if (onProgress) {
+							transferred += value.byteLength;
+							const percent = total === 0 ? 0 : transferred / total;
+
+							if (percent !== 1) {
+								onProgress(percent, transferred, total);
```

Comment:
I think we should also give the chunk here.

Reply:
Also I think we should remove `setInterval` in Got and emit progress events every new chunk like it's done here.

Reply:
> I think we should also give the chunk here.

Sure. Could be useful until we implement a proper stream API.

> Also I think we should remove setInterval in Got and emit progress events every new chunk like it's done here.

👍 If it's possible. There's must be a good reason we did the `setInterval` thing in the first place.

Reply:
> There's must be a good reason we did the setInterval thing in the first place, though.

The only reason I can think of is to not spam the progress events.